### PR TITLE
MYFACES-4461: Remove registerCdiProducers checks / Use CDI for implicit objects

### PR DESCRIPTION
**Pull request by volosied**
https://issues.apache.org/jira/browse/MYFACES-4461

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by volosied** [comment]
Making another update soon regarding implicit objects.

**Comment by volosied** [comment]
More details about the NPE in  FlowMyFacesCDIRequestTestCase.testFlow1_12 : 

ContextualStorage wasn't initialized when using CDI ( this error also occurs in 3.0 and likely 2.3, but I'll fix it in another issue) 

```
Caused by: java.lang.NullPointerException
	at org.apache.myfaces.flow.cdi.FlowScopeContext.get(FlowScopeContext.java:249)
	at org.jboss.weld.contexts.PassivatingContextWrapper$AbstractPassivatingContextWrapper.get(PassivatingContextWrapper.java:84)
	at org.jboss.weld.bean.ContextualInstanceStrategy$DefaultContextualInstanceStrategy.get(ContextualInstanceStrategy.java:100)
	at org.jboss.weld.bean.ContextualInstance.get(ContextualInstance.java:50)
	at org.jboss.weld.manager.BeanManagerImpl.getReference(BeanManagerImpl.java:675)
	at org.jboss.weld.module.web.el.AbstractWeldELResolver.lookup(AbstractWeldELResolver.java:107)
	at org.jboss.weld.module.web.el.AbstractWeldELResolver.getValue(AbstractWeldELResolver.java:90)
	at jakarta.el.CompositeELResolver.getValue(CompositeELResolver.java:62)
	at org.apache.el.parser.AstIdentifier.getValue(AstIdentifier.java:94)
	at org.apache.el.parser.AstValue.getTarget(AstValue.java:70)
	at org.apache.el.parser.AstValue.setValue(AstValue.java:196)
	at org.apache.el.ValueExpressionImpl.setValue(ValueExpressionImpl.java:263)
	at org.apache.myfaces.flow.FlowHandlerImpl.doAfterEnterFlow(FlowHandlerImpl.java:386)
	at org.apache.myfaces.flow.FlowHandlerImpl.transition(FlowHandlerImpl.java:278)
	at org.apache.myfaces.application.NavigationHandlerImpl.applyFlowTransition(NavigationHandlerImpl.java:381)
	at org.apache.myfaces.application.NavigationHandlerImpl.handleNavigation(NavigationHandlerImpl.java:316)
	at org.apache.myfaces.application.NavigationHandlerImpl.handleNavigation(NavigationHandlerImpl.java:115)
	at org.apache.myfaces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:122)
	at jakarta.faces.component.UICommand.broadcast(UICommand.java:65)
	at jakarta.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1233)
	at jakarta.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:414)
	at jakarta.faces.component.UIViewRoot._process(UIViewRoot.java:1705)
	at jakarta.faces.component.UIViewRoot.processApplication(UIViewRoot.java:915)
	at org.apache.myfaces.lifecycle.InvokeApplicationExecutor.execute(InvokeApplicationExecutor.java:43)
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:172)
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:125)
```

**Review by melloware — APPROVED** [review]

File: impl/src/main/java/org/apache/myfaces/flow/cdi/FlowScopeContext.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ public <T> T get(Contextual<T> bean, CreationalContext<T> creationalContext)
         List<String> activeFlowMapKeys = getStorageHolder(facesContext).getActiveFlowMapKeys(facesContext);
         for (String flowMapKey : activeFlowMapKeys)
         {
-            ContextualStorage storage = getContextualStorage(facesContext, false, flowMapKey);
+            ContextualStorage storage = getContextualStorage(facesContext, true, flowMapKey);
```
Comment (volosied) [inline impl/src/main/java/org/apache/myfaces/flow/cdi/FlowScopeContext.java:247]: Atom decided to fix all the whitespace errors. Here's the NPE fix. true  - createIfNotExist

File: impl/src/main/java/org/apache/myfaces/el/ELResolverBuilder.java [hunk 2]
```diff
@@ -159,35 +157,19 @@ protected boolean isReplaceImplicitObjectResolverWithCDIResolver(FacesContext fa
         BeanManager beanManager = CDIUtils.getBeanManager(facesContext);
         if (beanManager != null)
         {
-            FacesConfigBeanHolder holder = CDIUtils.get(beanManager, FacesConfigBeanHolder.class);
```
Comment (tandraschko) [inline impl/src/main/java/org/apache/myfaces/el/ELResolverBuilder.java:162]: like discussed in JIRA, i would not touch this for now
we should create a extra issue + discussion

**Pull request merged**